File: examples/scala/src/test/scala/GivenStage.scala
import com.tngtech.jgiven.annotation._
import com.tngtech.jgiven._

class GivenStage extends Stage[GivenStage] {

  @ProvidedScenarioState
  var someInt = 0
  
  @ProvidedScenarioState
  var anotherInt = 0
  
  def some_value(a:Int) = {
    someInt = a
    this
  }
  
  def another_value(b:Int) = {
    anotherInt = b
    this
  }
  
}

File: jgiven-junit5/src/test/java/com/tngtech/jgiven/junit5/test/GivenStage.java
package com.tngtech.jgiven.junit5.test;

import com.tngtech.jgiven.Stage;
import com.tngtech.jgiven.annotation.ExpectedScenarioState;
import com.tngtech.jgiven.annotation.ProvidedScenarioState;

public class GivenStage extends Stage<GivenStage> {

    @ProvidedScenarioState
    String someState;

    public GivenStage some_state() {
        someState = "Some State";
        return self();
    }

    public GivenStage some_action() {
        return self();
    }

    public GivenStage some_outcome() {
        return self();
    }

}
